watertap/ui/fsapi.py

```diff
@@ -438,41 +470,97 @@ def from_csv(self, file: Union[str, Path], flowsheet):
                 continue
             # build raw dict from values and header
             data = {k: v for k, v in zip(header, row)}
-            # evaluate the object in the flowsheet
-            try:
-                data["obj"] = eval(data["obj"], {"fs": flowsheet})
-            except Exception as err:
-                raise ValueError(f"Cannot find object in flowsheet: {data['obj']}")
-            # evaluate the units
-            norm_units = data["ui_units"].strip()
-            if norm_units in ("", "none", "-"):
-                data["ui_units"] = pyo.units.dimensionless
-            else:
-                try:
-                    data["ui_units"] = eval(norm_units, {"units": pyo.units})
-                except Exception as err:
-                    raise ValueError(f"Bad units '{norm_units}': {err}")
+            # parse into object
+            data["obj"] = self._parse_object_text(data["obj"], flowsheet)
+            data["ui_units"] = self._parse_units_text(data["ui_units"])
             # process boolean values (starting with 'is_')
             for k in data:
                 if k.startswith("is_"):
-                    v = data[k].lower()
-                    if v == "true":
-                        data[k] = True
-                    elif v == "false":
-                        data[k] = False
-                    else:
-                        raise ValueError(
-                            f"Bad value '{data[k]}' "
-                            f"for boolean argument '{k}': "
-                            f"must be 'true' or 'false' "
-                            f"(case-insensitive)"
-                        )
+                    data[k] = self._parse_boolean_text(data[k], flowsheet)
             # add parsed export
+            # TODO: Mark output variables as deferred
             self.add(data=data)
             num += 1
 
         return num
 
+    @staticmethod
+    def _parse_object_text(text: str, flowsheet: Any) -> Any:
+        """Evaluate the object represented by 'text' in the context of the flowsheet.
+
+        Args:
+            text: Object starts with prefix 'fs' and then is an arbitrary combination
+                  of attributes and indices to indicate a variable/etc. within the flowsheet.
+            flowsheet: Flowsheet object substituted for 'fs.'
+
+        Returns:
+            Pyomo block/var/etc. within the flowsheet for 'text'
+
+        Raises:
+            ValueError, if the object named by 'text' could not be found
+        """
+        _log.debug(f"eval text={text}")
+        try:
+            obj = eval(text, {"fs": flowsheet})
```

Comment:
Following @bknueven's suggestion in #1277, Pyomo's `find_component()` would be a safer option than `eval()`, with the important added benefit that we don't require the flowsheet to be called `fs` (as long as `text` contains the correct attribute name, of course).

I believe that using it on the underlying `ConcreteModel` instance (which IIRC you can get from the flowsheet object with `flowsheet.parent_block()`) would not even require changes to existing string-based flowsheet exports (i.e. the same string starting with `fs.` would work in both cases).

Reply:
I don't think safe is a real issue , but sure whatever

Reply:
I dropped the fs. actually